docs2/site/docs/getting-started/dependency-injection.md

```diff
@@ -65,6 +65,35 @@ public void ConfigureServices(IServiceCollection services)
 }
 ```
 
+To avoid having to register all of the individual graph types in your project, you can
```

Comment:
Do you know about [AddGraphTypes](https://github.com/graphql-dotnet/server/blob/319764047fcc248d5e4055f43b06ecfb2d79e972/src/Core/GraphQLBuilderExtensions.cs#L80) ? One line of code and all types are registered. It will be logical to tell here about this method, if we stopped limited by the functionality of the main package.

Reply:
And... I would like to move `IGraphQLBuilder` into `GraphQL.MicrosoftDI` 😄 . It will be right. Previously there was no necessary project, where this interface would take its place.

Reply:
```csharp
services.AddGraphTypes(); //  Add all types that implement IGraphType in the calling assembly
services.AddGraphTypes(typeof(MySchema)); // Add all types that implement IGraphType in the assembly which specified type belongs to
```

Reply:
Isn’t that in the server project? I don’t use the server project and usually have to rewrite similar code in all my projects. I also automatically add CLR type mappings for all automatically registered types. Does server do that? The issue is that some types must not be mapped so I have had to create a `DoNotRegister` attribute to prevent mapping it.

Reply:
We certainly can mention it in the docs though

Reply:
I do not want to create a situation when we write the same code in different places. I use the server project. I know that it has a lot of code to fix/refactor. Part of the code is generally worth transferring to another place. Now I see a good reason for this.

Reply:
> I also automatically add CLR type mappings for all automatically registered types. Does server do that?

It scans assembly for types implementing `IGraphType`. It `can` do more. Let's add missing features. I suggest to move all these classes/methods not specific to ASP.NET Core into GraphQL.MicrosoftDI.

>  The issue is that some types must not be mapped so I have had to create a DoNotRegister attribute to prevent mapping it.

Agree. It's easy to add too.

Reply:
👍  I can work on it.

Reply:
It will be a breaking change to the server project and a new feature for main project (4.1/4.2). I'm OK with fast major version changes in the server project.

Reply:
> 👍 I can work on it.

Well, I probably won't be.  However, I do agree that `IGraphQLBuilder` and its related methods should be moved to this repo, if you want to work on it.  If we do move it, the new methods should be reviewed and properly documented, and I just don't have the time for it right now.  Adding automatic CLR type mappings will actually be a separate feature, because it needs to operate on a schema instance rather than on a service collection.  And so it actually might not belong in the GraphQL.MicrosoftDI package at all.  Maybe I will do that part if I have time...

Reply:
Ok